Sources/Fuzzilli/Core/FuzzEngine.swift

```diff
@@ -18,3 +18,69 @@ public protocol FuzzEngine: ComponentBase {
     // Performs a single round of fuzzing using the engine.
     func fuzzOne(_ group: DispatchGroup)
 }
+
+extension FuzzEngine {
+    public func execute(_ program: Program, stats: inout ProgramGeneratorStats) -> ExecutionOutcome {
+        fuzzer.dispatchEvent(fuzzer.events.ProgramGenerated, data: program)
+
+        let execution = fuzzer.execute(program)
+
+        switch execution.outcome {
+            case .crashed(let termsig):
+                var code = program.code
+                code.append(Instruction(Comment(execution.stderr)))
+                let program = Program(with: code)
+                fuzzer.processCrash(program, withSignal: termsig, origin: .local)
+
+            case .succeeded:
+                fuzzer.dispatchEvent(fuzzer.events.ValidProgramFound, data: program)
+                if let aspects = fuzzer.evaluator.evaluate(execution) {
+                    fuzzer.processInteresting(program, havingAspects: aspects, origin: .local)
+                }
+                stats.producedValidSample()
+
+            case .failed(_):
+                if self.fuzzer.config.diagnostics {
+                    var code = program.code
+                    code.append(Instruction(Comment(execution.stdout)))
+                    let program = Program(with: code)
+                    fuzzer.dispatchEvent(fuzzer.events.InvalidProgramFound, data: program)
+                } else {
+                    fuzzer.dispatchEvent(fuzzer.events.InvalidProgramFound, data: program)
+                }
+                stats.producedInvalidSample()
+
+            case .timedOut:
+                fuzzer.dispatchEvent(fuzzer.events.TimeOutFound, data: program)
+                stats.producedInvalidSample()
+        }
+
+        return execution.outcome
+    }
+
+    /// Generate some basic Prefix such that samples have some basic types available.
+    public func generateProgramPrefix(mode: ProgramBuilder.Mode = .aggressive) -> Program {
```

Comment:
Do you even need the Mode here since the builder only runs a few trivial CodeGenerators? I guess you can always set it to `.conservative` since it shouldn't make a difference and the prefix should anyway be as correct as possible, since otherwise it would really ruin the MutationEngine's day as it uses it for many mutations in a row

Reply:
Yes I agree :)